changelog/@unreleased/pr-685.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
```

Comment:
Could also mention the benefits like:

* Faster to calculate git version
* Worktrees are now supported??? (maybe we can close #564 and #65?)

Reply:
This PR should close those worktree issues.

However we are still blocked on using worktrees because our publishing plugins (both internal and external) still use jgit because they depend on https://github.com/nebula-plugins/gradle-info-plugin.

Reply:
Thanks for the comments. I have updated the changelog and also added a test to show that git worktree is indeed now supported. https://github.com/palantir/gradle-git-version/pull/698